lib/active_merchant/billing/gateways/pay_hub.rb

```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
+        '4086' => "INVALID_MONTHLY_TYPE_CODE",
+        '4087' => "INVALID_MONTHLY_WEEK_DAYS_CODE",
+        '4088' => "INVALID_MONTHLY_DAYS_POSSION_CODE",
+        '4089' => "INVALID_MONTHLY_DAYS_CODE",
+        '4090' => "INVALID_START_DATE_CODE",
+        '4091' => "INVALID_SPECIFIC_DATES_CODE",
+        '4092' => "SPECIFIC_SAME_DATES_CODE",
+        '4093' => "INVALID_RECURRING_DATA_CODE",
+        '4094' => "END_BILL_DATE_BEFORE_START_CODE",
+        '4095' => "CARD_HOLDER_DATA_REQUIRED_CODE",
+        '4096' => "CARD_HOLDER_CODE_REQUIRED_CODE",
+        '4097' => "INVALID_STATUS_CODE",
+        '4098' => "UNABLE_TO_CHANGE_STATUS",
+        '4099' => "CARD_VALIDATION_FAILED_CODE",
+        '4100' => "RECURRING_CIS_FAILED_CODE",
+        '4101' => "UNABLE_BUILT_RECURRING_FILTER__CODE",
+        '4103' => "UNABLE_TO_BUILT_NEXT_BILL_CODE",
+        '4501' => "INVALID_PHONE_TYPE",
+        '4502' => "INVALID_PHONE_NUMBER",
+        '4503' => "INVALID_ZIP_CODE",
+        '4504' => "INVALID_STATE_CODE",
+        '4505' => "INVALID_INTERVAL_TYPE",
+        '4506' => "INVALID_END_DATE",
+        '4507' => "INVALID_BILL_WEEKLY_DAYS",
+        '4508' => "INVALID_YEAR_MONTH_DAYS",
+        '4509' => "INVALID_YEAR_BILL_ON_MONTH_NO",
+        '4510' => "INVALID_MONTHLY_BILL_TYPE",
+        '4511' => "INVALID_MDAYS_POSITION",
+        '4512' => "INVALID_MDAYS",
+        '4513' => "INVALID_MDAY_EACH_POSITIONS",
+        '4514' => "INCONSISTENT_SCHEDULE_FIELDS",
+        '4515' => "ACCOUNT_IS_NOT_ADMINISTRATOR",
+        '4516' => "INCONSISTENT_CARD_DATA_FIELDS",
+        '4517' => "WRONG_OAUTH_TOKEN_FOR_MERCHANT",
+        '4518' => "NO_MERCHANT_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4519' => "NO_SETUP_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4520' => "NO_VT_CLIENT_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4521' => "INVALID_CURRENCY_CODE",
+        '4522' => "INVALID_RECURRING_BILL_STATUS_CHANGE_CODE",
+        '4523' => "INCONSISTENT_CURRENCIES",
+        '4524' => "MISSING_PROPERTY_CODE",
+        '4525' => "INCONSISTENT_CUSTOMER_DATA_CODE",
+        '4526' => "DUPLICATE_DEVELOPER_CODE",
+        '9989' => "METHOD_REQUEST_BODY_VALIDATION_ERROR_CODE",
+        '9990' => "HTTP_NOT_READABLE_EXCEPTION_CODE",
+        '9991' => "JSONPARSE_EXCEPTION_CODE",
+        '9992' => "MAPPING_EXCEPTION_CODE",
+        '9993' => "MEDIA_TYPE_NOT_SUPPORTED_CODE",
+        '9994' => "METHOD_NOT_SUPPORTED_CODE",
+        '9995' => "PAGE_NOT_FOUND_ERROR_CODE",
+        '9996' => "ACCESS_DENIED_ERROR_CODE",
+        '9997' => "JSON_SYNTAX_ERROR_CODE",
+        '9998' => "NOT_FOUND_ERROR_CODE",
+        '9999' => "INTERNAL_SERVER_ERROR_CODE"
       }
 
       def initialize(options={})
-        requires!(options, :orgid, :username, :password, :tid)
-
+        requires!(options, :organization_id, :oauth_token, :terminal_id, :version)
         super
       end
-
-      def authorize(amount, creditcard, options = {})
```

Comment:
@j-mutter

Yes correct, previously I was using `authorize`  but now asp er new changes we validates through oauth_token in headers.  You can see in [headers](https://github.com/grepruby/active_merchant/blob/payhub_updates/lib/active_merchant/billing/gateways/pay_hub.rb#L389) method
        

      def headers(options = {})
        oauth_token = options[:oauth_token] || @options[:oauth_token]

        headers = {
          "Authorization" => "Bearer " + oauth_token,
          "Content-Type" => "application/json",
          "Accept" => "application/json"
        }
        headers
      end